library/src/interactor_impl.cxx

```diff
@@ -340,15 +341,15 @@ class interactor_impl::internals
           auto now = start;
           while (now < end)
           {
-            const double linear_t =
+            const double t =
               std::chrono::duration_cast<std::chrono::milliseconds>(now - start).count() /
               (double)self->TransitionDuration;
-            const double tween_t = (1 - std::cos(vtkMath::Pi() * linear_t)) / 2;
+            const double u = (1 - std::cos(vtkMath::Pi() * t)) / 2;
 
             self->Window.getCamera().setFocalPoint(
-              { foc[0] + dx * tween_t, foc[1] + dy * tween_t, foc[2] + dz * tween_t });
+              { foc[0] + dx * u, foc[1] + dy * u, foc[2] + dz * u });
             self->Window.getCamera().setPosition(
-              { pos[0] + dx * tween_t, pos[1] + dy * tween_t, pos[2] + dz * tween_t });
+              { pos[0] + dx * u, pos[1] + dy * u, pos[2] + dz * u });
```

Comment:
If the picked point is closer or further, there is a side effect of producing a zoom.  
I think this is because you are applying the same vector to the focal point and the position.  
I think it can be fixed easily by projecting the new position on the camera plane.
In practice, it's just a dot product. Here's a sketch to illustrate what I'm suggesting, let me know if you need help implementing it.

![image](https://user-images.githubusercontent.com/6069752/203503542-66466777-6f8d-43c5-bbd0-8738a73cdd0d.png)

Reply:
If I understand correctly, on the following sketch, when focusing from `f0` to `f1`, the camera should go form `p0` to `p2` instead of `p1`, is that right?
![focal-move](https://user-images.githubusercontent.com/489715/203510990-1463311b-f7ce-43b4-ac4c-5127eadc52bb.jpg)

Reply:
Exactly. I think it makes more sense. What do you think?

Reply:
The camera will travel in the plane and therefore will not go back and forth (which sounds nice) but the distance to the focal point will not be constant anymore (which could backfire?). This new method might actually produce more of a "zoom" effect (getting closer/further to the object) while the old one has more of a "dolly" effect but maintains the same distance.

Reply:
The focal point is used only to compute the camera direction and is used as the rotation center, so I don't think it will affect the zoom factor.  
The "dolly" effect you are mentioning is what I'd like to avoid, but maybe it's just my preference, I'm not pushing for it but it would be great if we can compare the two behaviors and decide what makes more sense.

Reply:
@snoyer I quickly tested and it behaves better now in my opinion. So I vote to keep this new version :)

Reply:
Agreed, I haven't found a situation where the varying distance would feel worse than flying along the model when picking far away. Good call.